src/core/dbus-execute.c

```diff
@@ -980,12 +981,20 @@ static int property_get_exec_dir_symlink(
         if (r < 0)
                 return r;
 
-        FOREACH_ARRAY(i, d->items, d->n_items)
+        FOREACH_ARRAY(i, d->items, d->n_items) {
+                /* The old exec directory properties cannot represent flags, so list them here with no
+                 * destination */
+                if ((i->flags & _EXEC_DIRECTORY_FLAGS_PUBLIC) != 0 && !strv_isempty(i->symlinks)) {
+                        r = sd_bus_message_append(reply, "(sst)", i->path, "", (uint64_t) (i->flags & _EXEC_DIRECTORY_FLAGS_PUBLIC));
+                        if (r < 0)
+                                return r;
+                }
```

Comment:
i figure this would be more readable if this was written like this:

```c
FOREACH_ARRAY(i, d->items, d->n_items) {
        if (strv_isempty(…)) { 
                /* generate empty symlink entry */
        } else {
                /* generate one entry for each symlink */
        } 
``